flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/calls/CurrentTimePointCallGen.scala

```diff
@@ -31,28 +32,28 @@ class CurrentTimePointCallGen(
   extends CallGenerator {
 
   override def generate(
-      codeGenerator: CodeGenerator,
-      operands: Seq[GeneratedExpression])
-    : GeneratedExpression = targetType match {
+      ctx: CodeGeneratorContext,
```

Comment:
Use `ExprCodeGenerator` here such that we don't need to pass `nullCheck` to every util function, if those functions are part of `ExprCodeGenerator` class. With this approach we don't need to add `nullCheck` to `CallGenerate.generate`.